packages/flutter_tools/lib/src/web/module_metadata.dart

```diff
@@ -0,0 +1,189 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Module metadata format version
+///
+/// Module reader always creates the current version but is able to read
+/// metadata files with later versions as long as the changes are backward
+/// compatible, i.e. only minor or patch versions have changed.
+class ModuleMetadataVersion {
+  const ModuleMetadataVersion(this.majorVersion, this.minorVersion, this.patchVersion);
+
+  final int majorVersion;
+  final int minorVersion;
+  final int patchVersion;
+
+  /// Current metadata version
+  ///
+  /// Version follows simple semantic versioning format 'major.minor.patch'
+  /// See https://semver.org
+  static const ModuleMetadataVersion current = ModuleMetadataVersion(2, 0, 0);
+
+  /// Previous version supported by the metadata reader
+  static const ModuleMetadataVersion previous = ModuleMetadataVersion(1, 0, 0);
+
+  /// Current metadata version created by the reader
+  String get version => '$majorVersion.$minorVersion.$patchVersion';
+
+  /// Is this metadata version compatible with the given version
+  ///
+  /// The minor and patch version changes never remove any fields that current
+  /// version supports, so the reader can create current metadata version from
+  /// any file created with a later writer, as long as the major version does
+  /// not change.
+  bool isCompatibleWith(String version) {
+    final List<String> parts = version.split('.');
+    if (parts.length != 3) {
+      throw FormatException(
+        'Version: $version'
+        'does not follow simple semantic versioning format',
+      );
+    }
+    final int major = int.parse(parts[0]);
+    final int minor = int.parse(parts[1]);
+    final int patch = int.parse(parts[2]);
+    return major == majorVersion && minor >= minorVersion && patch >= patchVersion;
+  }
+}
+
+/// Library metadata
+///
+/// Represents library metadata used in the debugger,
+/// supports reading from and writing to json.
+class LibraryMetadata {
+  LibraryMetadata(this.name, this.importUri, this.partUris);
+
+  LibraryMetadata.fromJson(Map<String, dynamic> json)
+    : name = _readRequiredField(json, 'name'),
```

Comment:
We should have constants defined for keys that are referenced in multiple places.

Reply:
Done.